compiler/src/main/java/com/squareup/anvil/compiler/ModuleMerger.kt

```diff
@@ -288,13 +258,25 @@ internal class ModuleMerger(
       }
   }
 
+  private fun Sequence<ClassReference>.types(codegen: ImplementationBodyCodegen): Set<Type> {
+    return cast<Sequence<ClassReference.Descriptor>>()
+      .map { codegen.typeMapper.mapType(it.clazz) }
+      .toSet()
+  }
+
+  private fun List<ClassReference>.types(codegen: ImplementationBodyCodegen): Set<Type> {
+    return cast<List<ClassReference.Descriptor>>()
+      .map { codegen.typeMapper.mapType(it.clazz) }
+      .toSet()
+  }
```

Comment:
```suggestion
  private fun Set<ClassReference>.types(codegen: ImplementationBodyCodegen): Set<Type> {
    return cast<Set<ClassReference.Descriptor>>()
      .map { codegen.typeMapper.mapType(it.clazz) }
  }
```
I'd change it to avoid the duplication. With this change we'll also rely on the equals() implementation within ClassReference, which gives us more control. That we relied on the internal compiler implementation always made me a little uneasy.

Reply:
👍  updated

> With this change we'll also rely on the equals() implementation within ClassReference, which gives us more control. That we relied on the internal compiler implementation always made me a little uneasy.

Could you explain that part of the comment though? It's not clear to me why this change has an effect on equals() being used

Reply:
If you convert a list to a set, then duplicated entries in the list will removed and not added to the set. For that the runtime will call `equals()`. If you convert `List<ClassReference>` to a set, then the `equals` method for `ClassReference` will be used and not the method from the Kotlin compiler type. 

So originally it was:
```
List<ClassReference> -> map to List<Type> -> to set (uses equals from Type)
```
Now it's:
```
List<ClassReference> -> to set (uses equals from ClassReference) -> map to Set<Type> 
```

Reply:
got it, thanks!